src/aiida_quantumespresso/parsers/matdyn.py

```diff
@@ -3,27 +3,31 @@
 from qe_tools import CONSTANTS
 
 from aiida_quantumespresso.calculations.matdyn import MatdynCalculation
+from aiida_quantumespresso.utils.mapping import get_logging_container
 
-from .base import Parser
+from .base import BaseParser
 
 
-class MatdynParser(Parser):
-    """Parser implementation for the MatdynCalculation."""
+class MatdynParser(BaseParser):
+    """``Parser`` implementation for the ``MatDynCalculation`` calculation job class."""
 
     def parse(self, **kwargs):
-        """Parse the retrieved files from a `MatdynCalculation`."""
-        retrieved = self.retrieved
-        filename_stdout = self.node.get_option('output_filename')
-        filename_frequencies = MatdynCalculation._PHONON_FREQUENCIES_NAME
+        """Parse the retrieved files from a ``MatdynCalculation`` into output nodes."""
+        logs = get_logging_container()
+
+        _, parsed_data, logs = self.parse_stdout_from_retrieved(logs)
 
-        if filename_stdout not in retrieved.base.repository.list_object_names():
-            return self.exit(self.exit_codes.ERROR_OUTPUT_STDOUT_READ)
+        base_exit_code = self.check_base_errors(logs)
+        if base_exit_code:
+            return self.exit(base_exit_code, logs)
 
-        if 'JOB DONE' not in retrieved.base.repository.get_object_content(filename_stdout):
-            return self.exit(self.exit_codes.ERROR_OUTPUT_STDOUT_INCOMPLETE)
+        self.out('output_parameters', orm.Dict(parsed_data))
 
-        if filename_frequencies not in retrieved.base.repository.list_object_names():
-            return self.exit(self.exit_codes.ERROR_OUTPUT_STDOUT_READ)
```

Comment:
This is removed on purpose:

1. I'm don't think this _ever_ happens if the stdout is "complete"
2. It should use its own exit code, and I don't want to define one if [1] is true.